File: linguagem_dart/encapsulamento/academia/personal.dart
import 'academia.dart';
import 'aluno.dart';

class Personal {
  // Atributos da classe Personal
  String _nome = "";
  late Aluno _aluno;

  // Getter e setter de nome
  String get getNome => _nome;

  set setNome(String nome) {
    _nome = nome;
  }

  // Getter e setter de aluno
  Aluno get getAluno => _aluno;

  set setAluno(Aluno aluno) {
    _aluno = aluno;
  }

  // Construtor da classe Personal
  Personal(String nomeDoPersonal, Aluno alunoDoPersonal) {
    setNome = nomeDoPersonal;
    setAluno = alunoDoPersonal;
  }

  // Método que retorna as informações
  String retornarInformacoesPersonal(Academia localDeTrabalho) {
    return "\nO(a) personal $getNome acompanha o(a) aluno(a) ${_aluno.getNome} na academia   ${localDeTrabalho.getNome}";
  }
}

File: linguagem_java/encapsulamento/academia/Personal.java
package encapsulamento.academia;

public class Personal {
    // Atributos da classe Personal
    private String nome;
    private Aluno aluno;

    // Construtor da classe Personal
    public Personal(String nomeDoAluno, Aluno alunoDoPersonal) {
        setNome(nomeDoAluno);
        setAluno(alunoDoPersonal);
    }

    // Getter e setter de nome
    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    // Getter e setter do aluno
    public Aluno getAluno() {
        return aluno;
    }

    public void setAluno(Aluno aluno) {
        this.aluno = aluno;
    }

    // Método para retornar informações
    public String retornarInformacoesPersonal(Academia localDeTrabalho) {
        return "\nO personal " + getNome() + " acompanha o aluno " + getAluno().getNome() + " na academia "
                + localDeTrabalho.getNome();
    }

}
